infection team change switch for split screen fix

Kantanomo opened the pull request.

Kantanomo committed:
infection team change switch for split screen fix

nukeulater commented on xlive/H2MOD/Variants/Infection/Infection.cpp:
```diff
@@ -95,14 +95,18 @@ void Infection::sendTeamChange()
 						{
-							user_interface_controller_set_desired_team_index(_controller_index_0, team);
-							user_interface_controller_update_network_properties(_controller_index_0);
+							player_indexes[player_array_index] = i;
```
this is not really needed for local players, just the interface code

nukeulater replied:
also https://github.com/pnill/cartographer/blob/8e566ef2f8bded4e323536fcb00b69174f1a9402/xlive/H2MOD/Variants/Infection/Infection.cpp#L467 needs to be taken care of

Kantanomo replied:
it is needed or the session doesn't update properly, the local profile will change teams but any other players will still see them on the red team

nukeulater replied:
that shouldn't be an issue, there must be something else wrong

Kantanomo replied:
from all my testing it wont work without it.

Kantanomo replied:
removed

Kantanomo committed:
more zombies work

fixed all  sections of zombies that strictly referenced local user 0

refactored and re-did hud_draw_player_indicators, now can control the player indicator rendering per-user

added helper function to player globals to get controller_index from user index.

Kantanomo committed:
removing code deemed unneeded
